mlir/include/mlir/Dialect/Ptr/IR/PtrOps.td

```diff
@@ -118,6 +118,124 @@ def Ptr_PtrAddOp : Pointer_Op<"ptr_add", [
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// LoadOp
+//===----------------------------------------------------------------------===//
+
+def AlignmentProp : OptionalProp<I64Prop>;
+
+def Ptr_LoadOp : Pointer_Op<"load", [
+    DeclareOpInterfaceMethods<MemoryEffectsOpInterface>
+  ]> {
+  let description = [{
+    The `load` operation is used to read from memory. A load may be marked as
+    atomic, volatile, and/or nontemporal.
```

Comment:
We would need to document what these mean I think.

Reply:
Currently, all of these share the same semantics of the llvm load inst. What the LLVM dialect does is to refer users to the LLVM Lang Ref with a link. Would that be okay, or better to document then here?

Reply:
Yes. The comment can be more specific and say, that the flags have the same meaning and not the op. Because the meaning of many of those flags is not exclusive to LLVM.